src/library_int53.js

```diff
@@ -0,0 +1,103 @@
+mergeInto(LibraryManager.library, {
+  // Writes the given JavaScript Number to the WebAssembly heap as a 64-bit integer variable.
+  // If the given number is not in the range [-2^53, 2^53] (inclusive), then an unexpectedly
+  // rounded or incorrect number can be written to the heap. ("garbage in, garbage out")
+#if ASSERTIONS
+  $writeI53ToI64__deps: ['$readI53FromI64', '$readI53FromU64'
+#if MINIMAL_RUNTIME
+    , '$warnOnce'
+#endif
+  ],
+#endif
+  $writeI53ToI64: function(ptr, num) {
+    HEAPU32[ptr>>2] = num;
+    HEAPU32[ptr+4>>2] = (num - HEAPU32[ptr>>2])/4294967296;
+#if ASSERTIONS
+    var deserialized = (num >= 0) ? readI53FromU64(ptr) : readI53FromI64(ptr);
+    if (deserialized != num) warnOnce('writeI53ToI64() out of range: serialized JS Number ' + num + ' to Wasm heap as bytes lo=0x' + HEAPU32[ptr>>2].toString(16) + ', hi=0x' + HEAPU32[ptr+4>>2].toString(16) + ', which deserializes back to ' + deserialized + ' instead!');
+#endif
+  },
+
+  // Same as writeI53ToI64, but if the double precision number does not fit within the
+  // 64-bit number, the number is clamped to range [-2^63, 2^63-1].
+  $writeI53ToI64Clamped: function(ptr, num) {
+    if (num > 0x7FFFFFFFFFFFFFFF) {
+      HEAPU32[ptr>>2] = 0xFFFFFFFF;
+      HEAPU32[ptr+4>>2] = 0x7FFFFFFF;
+    } else if (num < -0x8000000000000000) {
+      HEAPU32[ptr>>2] = 0;
+      HEAPU32[ptr+4>>2] = 0x80000000;
+    } else {
+      HEAPU32[ptr>>2] = num;
+      HEAPU32[ptr+4>>2] = (num - HEAPU32[ptr>>2])/4294967296;
+    }
+  },
+
+  // Like writeI53ToI64, but throws if the passed number is out of range of int64.
+  $writeI53ToI64Signaling: function(ptr, num) {
+    if (num > 0x7FFFFFFFFFFFFFFF || num < -0x8000000000000000) {
+#if ASSERTIONS
+      throw 'RangeError in writeI53ToI64Signaling(): input value ' + num + ' is out of range of int64';
+#else
+      throw 'RangeError:'+num;
+#endif
+    }
+    HEAPU32[ptr>>2] = num;
+    HEAPU32[ptr+4>>2] = (num - HEAPU32[ptr>>2])/4294967296;
+  },
+
+  // Uint64 variant of writeI53ToI64Clamped. Writes the Number to a Uint64 variable on
+  // the heap, clamping out of range values to range [0, 2^64-1].
+  $writeI53ToU64Clamped: function(ptr, num) {
+    if (num > 0xFFFFFFFFFFFFFFFF) HEAPU32[ptr>>2] = HEAPU32[ptr+4>>2] = 0xFFFFFFFF;
+    else if (num < 0) HEAPU32[ptr>>2] = HEAPU32[ptr+4>>2] = 0;
+    else {
+      HEAPU32[ptr>>2] = num;
+      HEAPU32[ptr+4>>2] = (num - HEAPU32[ptr>>2])/4294967296;
+    }
+  },
+
+  // Like writeI53ToI64, but throws if the passed number is out of range of uint64.
```

Comment:
I64 in comment should be U64

Reply:
There is actually no `writeI53ToU64` function. The implementation of that function would be identical to `writeI53ToI64` modulo bounds checking, which is what is done in ASSERTIONS, and in the `*Clamped` functions. Added notes to highlight this fact.